pkg/ccl/backupccl/restore_job.go

```diff
@@ -923,10 +936,57 @@ func createImportingDescriptors(
 				return errors.Wrapf(err, "restoring %d TableDescriptors from %d databases", len(r.tables), len(databases))
 			}
 
+			b := txn.NewBatch()
+
+			// For new schemas with existing parent databases, the schema map on the
+			// database descriptor needs to be updated.
+			existingDBsWithNewSchemas := make(map[descpb.ID][]catalog.SchemaDescriptor)
+			for _, sc := range writtenSchemas {
+				parentID := sc.GetParentID()
+				if _, ok := newDatabaseIDs[parentID]; !ok {
```

Comment:
you could also look at the descRewriteMap if that is present

Reply:
Done.